src/fromager/bootstrapper.py

```diff
@@ -132,22 +145,55 @@ def bootstrap(self, req: Requirement, req_type: RequirementType) -> Version:
                     source_filename=source_filename,
                     version=resolved_version,
                 )
-                unpack_dir = sdist_root_dir.parent
             else:
                 sdist_root_dir = unpacked_cached_wheel
-                unpack_dir = unpacked_cached_wheel.parent
+
+            assert sdist_root_dir is not None
+            unpack_dir = sdist_root_dir.parent
 
             # need to call this function irrespective of whether we had the wheel cached
             # so that the build dependencies can be bootstrapped
             build_dependencies = self._prepare_build_dependencies(req, sdist_root_dir)
 
-            # skip building even if it is a non-fromager built wheel
-            if not cached_wheel_filename:
-                wheel_filename, build_env = self._build(
+            if cached_wheel_filename:
```

Comment:
I can imagine us wanting to understand why a wheel is being built or not based on these flags, especially if it is encountered multiple times and sometimes it is treated as an sdist and other times it is treated as a wheel. Could you add some log statements that will help us debug these different cases in this if:then block?

Reply:
Sure! That's a reasonable request.